cfgtool

```diff
@@ -233,11 +227,77 @@ The user's configuration has been created under ${client_vpn_tarname} .
 Please transmit the tar over a safe channel to it's user.
 
 In order to access the VPN the user must unpack the tar, make sure the 'nobody' group and user exists on their system and execute 'openvpn openvpn-client.cfg' inside the unpacked directory as root.
+  
+Stay safe. Use crypto. Don´t drink and factorize."
+
+}
+
+__add_and_tar_user() {
+  name="$1"; shift || fail "USAGE: ${script_file} add_user <name> <output_dir>"
+  tmp_dir=$1; shift || fail "USAGE: ${script_file} add_user <name> <output_dir>"
+  client_vpn_dirname="${tmp_dir}/vpn_${VPN_NAME}_${name}"
+  client_vpn_tarname="${client_vpn_dirname}.tar"
+
+  ${script_file} add_user $name $tmp_dir
+
+  cd "`dirname "$client_vpn_dirname"`"
+  tar -cvf "$client_vpn_tarname" "`basename "$client_vpn_dirname"`"
+
+  er "
+The user's configuration has been created under ${client_vpn_tarname} .
+
+Please transmit the tar over a safe channel to it's user.
+
+In order to access the VPN the user must unpack the tar, make sure the 'nobody' group and user exists on their system and execute 'openvpn openvpn-client.cfg' inside the unpacked directory as root.
+  
+Stay safe. Use crypto. Don´t drink and factorize."
+
+}
 
+__add_and_zip_user() {
+  name=$1;
+  password=$2;
+  tmp_dir=$3;
+  client_vpn_dirname="${tmp_dir}/vpn_${VPN_NAME}_${name}"
+  client_vpn_zipname="$vpn_${VPN_NAME}_${name}.zip"
+
+  ${script_file} add_user $name $tmp_dir
+  cd $client_vpn_dirname    
+  zip -P $password -r $client_vpn_zipname ./
+
+  er "
+The user's configuration has been created under ${client_vpn_zipname}.
+
+And encrypted with the generated password.
+
+Please transmit the zip over a safe channel to it's user.
+
+In order to access the VPN the user must unpack the tar, make sure the 'nobody' group and user exists on their system and execute 'openvpn openvpn-client.cfg' inside the unpacked directory as root.
+  
 Stay safe. Use crypto. Don´t drink and factorize."
 
 }
 
+load_clients() {
+  file_name="$1"; shift || fail "USAGE: ${script_file} load_clients <filename>"
+  IFS=$'\n' clients=($(<$file_name))
+  temp_dir="./temp_clients"
+  rm -f "generated_clients.txt"
+
+  for client in ${clients[@]}
+  do
+    name=$(echo $client | awk -F @ '{ print $1 }')
+    password=$(pwgen -s 80 -1 -v -c -0)
```

Comment:
oh fair eough Didn't know that.
I read it as that a new password gets generated for each client